core/src/main/scala/chisel3/IO.scala

```diff
@@ -23,7 +23,12 @@ object IO {
     */
   def apply[T <: Data](iodef: => T)(implicit sourceInfo: SourceInfo): T = {
     val module = Module.currentModule.get // Impossible to fail
-    require(module.isIOCreationAllowed, "This module cannot have user-created IO")
+    if (!module.isIOCreationAllowed)
+      Builder.error(
+        s"This module cannot have IOs instantiated after disallowing IOs: ${module._whereIOCreationIsDisallowed
+          .map(_.makeMessage { s: String => s })
+          .mkString(",")}"
+      )
```

Comment:
I didn't realize that creating the line-caret error messages were so easy. This looks like:

```
[info] - should be able to lock down their ios in a scope *** FAILED ***
[info]   chisel3.package$ChiselException: src/test/scala/chiselTests/FixedIOModuleSpec.scala 75:25: This module cannot have IOs instantiated after disallowing IOs: @[src/test/scala/chiselTests/FixedIOModuleSpec.scala 73:17]
[info]           val other = IO(Input(UInt(1.W)))
[info]                         ^
```